falcon/request.py

```diff
@@ -2066,6 +2077,16 @@ def get_param_as_list(
         if name in params:
             items = params[name]
 
+            # If a delimiter is specified AND the param is a single string, split it.
+            if delimiter and isinstance(items, str):
+                if delimiter == ' ':
+                    items = items.split(' ')
+                elif delimiter == '|':
+                    items = items.split('|')
+                else:
+                    # For commas and others, also strip whitespace for convenience.
+                    items = [v.strip() for v in items.split(delimiter)]
```

Comment:
I don't think we want undocumented side effects like stripping whitespace, moreover, depending on the value of the delimiter.